akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/RoutersSpec.scala

```diff
@@ -115,6 +113,48 @@ class RoutersSpec extends ScalaTestWithActorTestKit("""
       }
     }
 
+    "support broadcast" must {
+      trait Cmd
+      case object ReplyWithAck extends Cmd
+      case class BCast(cmd: Cmd) extends Cmd
+
+      def behavior(replyTo: ActorRef[AnyRef]) = Behaviors.setup[Cmd] { ctx =>
+        Behaviors.receiveMessage[Cmd] {
+          case ReplyWithAck =>
+            val reply = ctx.self.path
+            replyTo ! reply
+            Behaviors.same
+          case BCast(BCast(_)) =>
+            fail("totally unacceptable")
+          case BCast(ReplyWithAck) =>
+            val reply = ctx.self.path
+            replyTo ! reply
+            Behaviors.same
+        }
+      }
+
+      "when disabled" in {
+        val probe = testKit.createTestProbe[AnyRef]()
+        val pool = testKit.spawn(Routers.pool(4)(behavior(probe.ref)))
+        pool ! BCast(ReplyWithAck)
+        probe.expectMessageType[ActorPath]
+        probe.expectNoMessage()
+      }
```

Comment:
Maybe I was a bit unclear, what I meant was to cover that non-broadcast works for messages that the custom predicate returns false for. The default case should be covered already by other tests.

Reply:
👍 look at following commit